crates/red_knot_workspace/src/workspace/settings.rs

```diff
@@ -59,15 +67,19 @@ pub struct SearchPathConfiguration {
 }
 
 impl SearchPathConfiguration {
-    pub fn into_settings(self, workspace_root: &SystemPath) -> SearchPathSettings {
-        let site_packages = self.site_packages.unwrap_or(SitePackages::Known(vec![]));
+    pub fn to_settings(&self, workspace_root: &SystemPath) -> SearchPathSettings {
+        let site_packages = self
+            .site_packages
+            .clone()
+            .unwrap_or(SitePackages::Known(vec![]));
 
         SearchPathSettings {
-            extra_paths: self.extra_paths.unwrap_or_default(),
-            src_root: self
+            extra_paths: self.clone().extra_paths.unwrap_or_default(),
+            workspace_root: self
+                .clone()
                 .src_root
```

Comment:
So the `src_root` from the `SearchPathConfiguration` represents some explicit red-knot config where the user tells us their source root, and this overrides setting the discovered workspace root as the workspace root in the `SearchPathSettings`?

Reply:
We should change the behavior as we discussed it in our 1:1 but it's out of scope for this PR